File: backend/src/main/java/com/gss/inventory/inventory/domain/model/records/InventoryItem.java
package com.gss.inventory.inventory.domain.model.records;

import java.util.List;

public record InventoryItem(
    Long id,
    String name,
    List<Category> categories,
    String description,
    String location,
    int totalQuantity,
    int availableQuantity,
    List<Image> images
) {
    public record Category(Long id, String name, String label) {}
    public record Image(Long id, String url) {}
}

File: frontend/src/features/inventory/domain/inventoryItem.ts
export type ItemCategory = {
  id: number
  name: string
  label: string
  sortOrder: number
}

export type ItemImage = {
  id: number
  url: string
}

export type InventoryItem = {
  id: number
  name: string
  categories: ItemCategory[]
  description: string | null
  location: string | null
  totalQuantity: number
  availableQuantity: number
  images: ItemImage[]
}

export type InventoryItemDraft = {
  name: string
  categoryIds: number[]
  description: string
  location: string
  totalQuantity: number
}

export const emptyInventoryItemDraft: InventoryItemDraft = {
  name: '',
  categoryIds: [],
  description: '',
  location: '',
  totalQuantity: 1,
}
